src/Serializer/PropertyFactoryOptionsTrait.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Serializer;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Serializer\Normalizer\AbstractNormalizer;
+
+/**
+ * Manipulate the property factory options.
+ *
+ * @author @author Alan Poulain <contact@alanpoulain.eu>
+ */
+trait PropertyFactoryOptionsTrait
+{
+    /**
+     * Gets the options for a property factory.
+     */
+    private function getPropertyFactoryOptions(string $resourceClass, bool $splitDeserialization = false): array
+    {
+        $serializationContext = $this->serializerContextBuilder->createFromRequest(Request::createFromGlobals(), true, [
```

Comment:
This isn't reliable: super globals aren't available in CLI, and the documentation is available in CLI. We should use a new interface not depending of `Request` as suggested in the comments.

Also, the result of `Request::createFromGlobals()` should be stored in a variable to prevent calling the method t two times.

Reply:
Actually, we don't really care of the data given by the request. It's only a workaround to make it work.

Reply:
Yes but it's ugly... and can be avoided.